net.longturn.freeciv21.yaml

```
@@ -0,0 +1,89 @@
+  - --share=network
+  - --socket=pulseaudio
+  - --filesystem=home
+
+command: freeciv21-client
+modules:
+  # Get Lua 5.3 as a dependency not included in the org.kde.Platform package.
+  - name: Lua
+    sources:
+      - type: archive
+        url: https://www.lua.org/ftp/lua-5.3.6.tar.gz
+        sha256: fc5fd69bb8736323f026672b1b7235da613d7177e72558893a0bdcd320466d60
+        dest: include/lua5.3
+    buildsystem: simple
+    build-commands:
+      - cd include/lua5.3 && make linux
+      - cp -rv * ${FLATPAK_DEST}
+    cleanup:
+      - '*'
+
+  # Get Libertunus font as a dependency.
+  - name: Libertinus
+    sources:
+      - type: archive
+        url: https://github.com/alerque/libertinus/releases/download/v7.040/Libertinus-7.040.zip
+        sha256: 2cce08507441d8ae7b835cfe51fb643ad5d9f6b44db4360c4e244f0e474a72f6
+        dest: fonts/Libertinus
+    buildsystem: simple
+    build-commands:
+      - mkdir -p ${FLATPAK_DEST}/share/freeciv21/fonts/Libertinus/static/OTF
+      - cp -rv fonts/Libertinus/static/OTF/*.otf ${FLATPAK_DEST}/share/freeciv21/fonts/Libertinus/static/OTF
+
+  # Now build and package freeciv21
+  - name: freeciv21
+    buildsystem: cmake-ninja
+    config-opts:
+      - -DCMAKE_BUILD_TYPE=Release
+      - -DFREECIV_DOWNLOAD_FONTS=OFF
+      - -DLUA_INCLUDE_DIR=/app/include/lua5.3/src
+      - -DLUA_LIBRARY=/app/include/lua5.3/src/liblua.a
+    cleanup:
+      - /include
+    sources:
+      - type: git
+        tag: v3.0-patch.1
+        url: https://github.com/longturn/freeciv21
+
+  # Finalize
+  - name: Finalize
+    buildsystem: simple
+    build-commands:
+      - mkdir -p ${FLATPAK_DEST}/share/icons/hicolor/64x64/apps
+      - cp ${FLATPAK_DEST}/share/freeciv21/misc/civicon.png ${FLATPAK_DEST}/share/icons/hicolor/64x64/apps/${FLATPAK_ID}.png
+      - cp ${FLATPAK_DEST}/share/freeciv21/misc/mpicon.png ${FLATPAK_DEST}/share/icons/hicolor/64x64/apps/net.longturn.freeciv21.modpack-qt.png
+      - mv ${FLATPAK_DEST}/share/applications/net.longturn.freeciv21.client.desktop ${FLATPAK_DEST}/share/applications/${FLATPAK_ID}.desktop
+      - sed -i 's/\/app\/share\/freeciv21\/misc\/civicon/net.longturn.freeciv21/' ${FLATPAK_DEST}/share/applications/${FLATPAK_ID}.desktop
+      - sed -i 's/\/app\/share\/freeciv21\/misc\/mpicon/net.longturn.freeciv21.modpack-qt/' ${FLATPAK_DEST}/share/applications/net.longturn.freeciv21.modpack-qt.desktop
+      - sed -i 's/\/app\/share\/freeciv21\/misc\/server/net.longturn.freeciv21/' ${FLATPAK_DEST}/share/applications/net.longturn.freeciv21.server.desktop
+      - sed -i 's/\/app\/share\/freeciv21\/misc\/civicon/net.longturn.freeciv21/' ${FLATPAK_DEST}/share/applications/net.longturn.freeciv21.ruledit.desktop
+      - mv ${FLATPAK_DEST}/share/metainfo/net.longturn.freeciv21.client.metainfo.xml ${FLATPAK_DEST}/share/metainfo/${FLATPAK_ID}.metainfo.xml
+      - sed -i 's/net.longturn.freeciv21.client/net.longturn.freeciv21/' ${FLATPAK_DEST}/share/metainfo/${FLATPAK_ID}.metainfo.xml
```

please 
```suggestion
      - appstream-util modify $FLATPAK_DEST/share/metainfo/net.longturn.freeciv21.client.metainfo.xml id ${FLATPAK_ID}",
```

this granted you use `rename-appdata-file` as mentionned above

---

`appstream-util` is also a cool tool.